docs/LwM2M_Client/Nordic/Runtime_certificate.md

```diff
@@ -0,0 +1,103 @@
+
+In **Certificate Mode** an asymmetrical public-key cryptographic algorithm is used to authenticate the connection endpoints and initialize payload encryption. Appropriate certificates need to be generated for both the LwM2M Client and the LwM2M Server. Public certificates of both parties are mutually available, and each party also has access to its corresponding private key.
+
+Read more about secure communication on [Anjay's documentation](https://avsystem.github.io/Anjay-doc/BasicClient/BC-Security.html) page.
+
+## Prerequisites
+* The nRF9160 board with a USB cable.
+- Cloned [Anjay-zephyr-client](https://github.com/AVSystem/Anjay-zephyr-client#getting-started) repository
+- Installed **nrfjprog** from [Nordic Semiconductor page](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools/download)
+* Installed **minicom** (for Linux), **RealTerm**, **PuTTy** (for Windows), or another serial communication program.
+* An active [Coiote IoT DM](https://eu.iot.avsystem.cloud/) user account.
+
+## Build and flash the device
+0. Connect the nRF9160 board to a USB port of your machine.
+0. Go to your local **Anjay-zephyr-client** directory
+0. Set West manifest path to `Anjay-zephyr-client/demo`, manifest file to `west-nrf.yml`, and do `west update`:
+
+    ```
+    west config manifest.path Anjay-zephyr-client/demo
+    west config manifest.file west-nrf.yml
+    west update
+    ```
+
+0. Go to the **Anjay-zephyr-client/demo** folder and build a project with a runtime certificate and private key.
+
+    ```
+    west build -b <BOARD> -p -- -DCONFIG_ANJAY_ZEPHYR_RUNTIME_CERT_CONFIG=y
+    ```
+
+    where <BOARD\> should be replaced by the selected board from `boards/` directory.
+
+    !!! Note
+
+        The runtime certificate and private key do not work with Thingy:91. Other boards should work with this command.
+
+    This feature works with nrf9160dk starting from revision v0.14.0. For this board use configuration that utilizes an external flash chip and software-based cryptography:
+
+    ```
+    west build -b nrf9160dk_nrf9160_ns@0.14.0 -p -- -DCONF_FILE=prj_extflash.conf -DOVERLAY_CONFIG="overlay_nrf_mbedtls.conf"
+    ```
+
+0. Find the merged.hex file under the `build/zephyr` directory in the project folder.
+
+0. Use the nRF Connect Programmer with the downloaded `.hex` file and execute steps from the [Updating the application firmware](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf9160_gs.html#updating-the-application-firmware) section.
+
+## Generate certificate
+The certificate and private key based on the SECP256R1 curve can be provided through the shell interface in `.pem` format. To generate them open terminal and use the following commands.
+
+!!! Important
+
+    To use the certificate and private key with Coiote IoT DM you must specify a common name that is the same as the client endpoint name.
```

Comment:
Where do I specify that? I've never done that and seriously I have no idea.

Reply:
by this commands. You create the certificate by typing informations in the terminal.

Reply:
But where is the place where I provide the `common name that is the same as the client endpoint name`., Your endpoint name here is the `anjay-zephyr-demo` but I don't see it in the example commands etc. I'm seriously confused.